src/v/cluster/controller.cc

```diff
@@ -537,10 +542,29 @@ ss::future<> controller::cluster_creation_hook(cluster_discovery& discovery) {
         vassert(
           config::node().seed_servers().empty(), "We are not on the root node");
 
-        if (
-          _raft0->last_visible_index() > model::offset{}
-          || _raft0->config().brokers().size() > 1) {
+        if (controller_log_exists()) {
             // The controller log has already been written to
+            if (!_storage.local().get_cluster_uuid()) { // HERE: ctrllog done?
```

Comment:
Why are we doing this in the cluster creation hook? And why are we doing this _if_ we're a founder. I thought the idea was that even existing clusters will need to run this upon upgrade.

Reply:
Great observation, however it boils down to the meaning of "cluster founder". The way it works now, the root is a cluster founder until cluster_uuid is created. It makes sense to change that (it's confusing to have a founder for an existing, although non-upgraded cluster); then this block should go to the very beginning of the `cluster_creation_hook`. Do you think this change is good for the scope of this PR?

Reply:
I suppose it's worth changing. I see in `cluster_discovery` we only check the local storage `cluster_uuid` in seeds-driven mode. FWIW though, now we check if there's a controller directory, so we probably wouldn't end up getting here anyway.

I'm not sure I understand why that's relevant though. It doesn't matter whether we're a even a seed server or not -- we should be spinning this fiber on every node, right? We have no guarantee that any of the seed servers will even become leader after restarting.